core/include/aoclient.h

```diff
@@ -1905,6 +1905,15 @@ class AOClient : public QObject {
      */
     bool checkPasswordRequirements(QString username, QString password);
     
+    /**
+     * @brief Checks if a testimony contains '<' or '>'.
+     *
+     * @param message The IC Message that might contain unproper symbols.
+     *
+     * @return True if it contains '<' or '>' symbols, otherwise false.
+     */
+
+    bool checkTestimonySymbols(QString message);
```

Comment:
You could make the parameter a const reference, and the function itself const as well, like so:

```suggestion
    bool checkTestimonySymbols(const QString& message) const;
```

For the parameter, making it a reference is uuuh, pretty much just a looks thing? Like, it has a benefit in that you may be able to swap out a _**thicc**_ move with just an indirection and a fixed sized thing, but let's not kid ourselves, you're developing on AO, so who really gives a shit.

That aside, `const` ensures that the parameter cannot be modified inside the function -- and in fact, that is what you want to (not) do.

If you check `.contains()`'s [documentation](https://doc.qt.io/qt-5/qstring.html#contains), you can see that it is a `const` member function of QString. A `const` member function essentially "promises" that it will not modify the class instance it is a member of. So `.contains()` will not change the `QString` you call it on.

Which does lead me into my second point, making the function itself `const` (that's the `const` at the end).
Since all you're doing is just checking a parameter's `const` function, and then returning a true/false based on that, you're not modifying the AOClient itself. So, you can limit yourself and later programmers this way.

I prefer keeping functions and parameters the most locked down at the start, so consts and refs everywhere, then "unlocking" stuff as I need 'em. Forces ya to think through when you *really* need to change shit.